Source/Core/CesiumTerrainProvider.js

```diff
@@ -629,7 +635,13 @@ define([
             extensionList.push('watermask');
         }
 
-        var promise = loadArrayBuffer(url, getRequestHeader(extensionList), request);
+        var resource = this._resource.getDerivedResource({
+            url: urlTemplates[(x + tmsY + level) % urlTemplates.length].replace('{z}', level).replace('{x}', x).replace('{y}', tmsY),
```

Comment:
URL template interpolation might be another good candidate to move into `Resource`, since we do this in a number of places.  Basically it could accept a templated URL and a dictionary of key-value parameters.